File: backend/src/main/java/com/mini/credit/dto/caisse/SessionCaisseOpeningContextResponse.java
package com.mini.credit.dto.caisse;

import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Builder
public class SessionCaisseOpeningContextResponse {
    private Long caisseId;
    private LocalDate dateComptable;
    private String devise;
    private boolean premiereSession;
    private BigDecimal soldeOuvertureAutomatique;
    private boolean forcageAutorise;
    private boolean soldeVerrouille;
    private boolean sessionExistante;
    private Long sessionExistanteId;
    private String sessionExistanteStatut;
    private java.time.LocalDateTime sessionExistanteDateOuverture;
    private java.time.LocalDateTime sessionExistanteDateCloture;
    private Long sessionExistanteUtilisateurId;
    private String sessionExistanteUtilisateurNom;
}

File: frontend/src/app/features/caisse/models/session-caisse-opening-context-response.ts
export interface SessionCaisseOpeningContextResponse {
  caisseId: number;
  dateComptable: string;
  devise: string;
  premiereSession: boolean;
  soldeOuvertureAutomatique: number;
  forcageAutorise: boolean;
  soldeVerrouille: boolean;
  sessionExistante?: boolean;
  sessionExistanteId?: number;
  sessionExistanteStatut?: string;
  sessionExistanteDateOuverture?: string;
  sessionExistanteDateCloture?: string;
  sessionExistanteUtilisateurId?: number;
  sessionExistanteUtilisateurNom?: string;
}
